From 69c62efb33e69ce542633aed173fddbfbd924de3 Mon Sep 17 00:00:00 2001
From: Byron Lee <byronlee@google.com>
Date: Mon, 28 Oct 2024 04:33:46 +0000
Subject: [PATCH] Revert "Remove deprecated base::GetFileSize() that uses
 out-parameter."

This reverts commit e6604bec9f9c7dde579ab21f1ec6721d7bb01ee3

Change-Id: Ib3cd24d5abdc89f2b5ef7ce20364e6c8bb4eadcc
---
 base/files/file_util.cc | 9 +++++++++
 base/files/file_util.h  | 4 ++++
 2 files changed, 13 insertions(+)

diff --git a/base/files/file_util.cc b/base/files/file_util.cc
index f85fda2550..b39f7a3082 100644
--- a/base/files/file_util.cc
+++ b/base/files/file_util.cc
@@ -423,6 +423,15 @@ OnceCallback<std::optional<int64_t>()> GetFileSizeCallback(
   return BindOnce([](const FilePath& path) { return GetFileSize(path); }, path);
 }
 
+bool GetFileSize(const FilePath& file_path, int64_t* file_size) {
+  std::optional<int64_t> maybe_size = GetFileSize(file_path);
+  if (!maybe_size.has_value()) {
+    return false;
+  }
+  *file_size = maybe_size.value();
+  return true;
+}
+
 bool TouchFile(const FilePath& path,
                const Time& last_accessed,
                const Time& last_modified) {
diff --git a/base/files/file_util.h b/base/files/file_util.h
index 0cb1885cd3..ab13211f75 100644
--- a/base/files/file_util.h
+++ b/base/files/file_util.h
@@ -484,6 +484,10 @@ BASE_EXPORT std::optional<int64_t> GetFileSize(const FilePath& file_path);
 BASE_EXPORT OnceCallback<std::optional<int64_t>()> GetFileSizeCallback(
     const FilePath& path);
 
+// Returns true on success and writes the file size to `file_size`. Otherwise
+// returns false and `file_size` is unmodified.
+BASE_EXPORT bool GetFileSize(const FilePath& file_path, int64_t* file_size);
+
 // Sets |real_path| to |path| with symbolic links and junctions expanded.
 // On Windows, the function ensures that the resulting |real_path| starts with a
 // drive letter.
-- 
2.47.0.163.g1226f6d8fa-goog

